gossip/server.go

```diff
@@ -125,7 +125,10 @@ func (s *server) jitteredGossipInterval() time.Duration {
 // then begins processing connecting clients in an infinite select
 // loop via goroutine. Periodically, clients connected and awaiting
 // the next round of gossip are awoken via the conditional variable.
-func (s *server) start(rpcServer *rpc.Server) {
+func (s *server) start(rpcServer *rpc.Server, stopper *util.Stopper) {
+	stopper.AddWorker()
+	defer stopper.SetStopped()
```

Comment:
How is it racy? This is entirely safe usage unless I'm missing something.

I tried doing it before the call and missed a bunch of cases in unittests which made me reconsider. Better to follow the same pattern everywhere which I've decided should be this:

    func (f *Foo) Start(stopper util.Stopper) {
      stopper.AddWorker()
      go func() {
        defer stopper.SetStopped()
        ...
      }()
    }